tests/main/refresh-app-awareness/task.yaml

```diff
@@ -15,17 +15,40 @@ restore: |
     # TODO: There is currently no way to unset configuration options.
     # Once this is fixed please uncomment this line:
     # snap unset core experimental.refresh-app-awareness
+    rm -f install.log
 execute: |
     # Install v1 and see that it runs as expected.
     snap install --dangerous test-snapd-refresh_1_all.snap
     test-snapd-refresh.version | MATCH v1
-    # Run a sleeper app to keep the snap busy
-    test-snapd-refresh.sh -c 'exec sleep 1h' &
+
+    # Run a sleeper app to keep the snap busy. The purpose of the stamp file is
+    # to allow us to synchronize with the concurrently running program.
+    #shellcheck disable=SC2016
+    test-snapd-refresh.sh -c 'touch $SNAP_DATA/stamp && exec sleep 1h' &
     pid=$!
     trap 'kill '"$pid"' || true' EXIT
-    # Try to install v2, it should fail because v1 is running. Snapd is even kind enough to tell us what is preventing the install from working.
-    ! snap install --dangerous test-snapd-refresh_2_all.snap | MATCH 'error: cannot install snap file: snap "test-snapd-refresh" has running apps \(sh\)'
+
+    # Ensure that snap-confine has finished its task and that the snap process
+    # is active. Note that we don't want to wait forever either.
+    wait_for_stamp() {
+        for _ in $(seq 30); do
+            test -e "$1" && break
+            sleep 0.1
+        done
+    }
+
+    wait_for_stamp /var/snap/test-snapd-refresh/current/stamp
+
+    # Try to install v2, it should fail because v1 is running. Snapd is kind
+    # enough to tell us what is preventing the install from working.
+    # The error message is automatically wrapped so unwrap it here.
+    unwrap_msg() {
```

Comment:
IIRC there's `string_to_one_line` in `tests/lib/strings.sh`. Not a blocker through.

Reply:
Interesting, I will send a follow up after this; I want to unbreak master.